src/ModelContextProtocol/Server/IMcpServerPrimitive.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace ModelContextProtocol.Server;
+
+/// <summary>Represents an MCP server primitive, like a tool or a prompt.</summary>
+public interface IMcpServerPrimitive
```

Comment:
I presume primitive is MCP lingo for a resource?

Reply:
We've struggled to come up with a unifying name for the things that are tools, prompts, resources, etc. The spec doesn't use the word primitive, but it shows up in places like this:
https://spec.modelcontextprotocol.io/specification/2024-11-05/architecture/#servers
"Expose resources, tools and prompts via MCP primitives"
I'm happy to subsequently change this to a better word if we can find a better one.
@PederHP